src/main/java/com/flightstats/hub/webhook/InternalWebhookClient.java

```diff
@@ -45,16 +45,40 @@ boolean stop(String name, String server) {
         return put(server + "/internal/webhook/stop/" + name);
     }
 
-    Optional<String> runOnServerWithFewestWebhooks(String name) {
-        return runOnOneServer(name, getOrderedServers());
+    Optional<String> runOnOnlyOneServer(String name, Collection<String> runningServers) {
+        Optional<String> successfulServerRun = Optional.ofNullable(
+                runOnOneServer(name, runningServers).orElseGet(
+                        () -> runOnServerWithFewestWebhooksExcluding(name, runningServers).orElse(null)));
+
+        runningServers.stream()
+                .filter(server -> successfulServerRun
+                        .map(successfulRun -> !server.equals(successfulRun))
+                        .orElse(true))
+                .forEach(server -> stop(name, server));
+
+        return successfulServerRun;
     }
 
     Optional<String> runOnOneServer(String name, Collection<String> servers) {
-        return servers.stream()
+        return runOnOneServer(name, servers.stream());
```

Comment:
This method name is same as another private method. Initially i thought this is recursive function and looking for terminating condition. Is it possible to rename one of them?

Reply:
One takes a collection and one takes a stream, the behavior is otherwise exactly the same, which seems like it'd be a reasonable candidate for argument overloading... If you feel strongly otherwise, I'll see about changing it.